hstool/templates/tool/relation_add.html

```diff
@@ -6,34 +6,54 @@
   <script>
     $(window).load(function() {
       $(':checkbox').change(function () {
-          var driver_checkbox = document.getElementById("id_check_driver");
-          var figureindicator_checkbox = document.getElementById("id_check_figureindicator");
-
-          $(':checkbox').not(this).prop('checked', false);
-          if (driver_checkbox.checked) {
-              $('[data-show-if=show-drivers]').removeClass('hide');
-              $('[data-show-if=show-figureindicators]').addClass('hide');
-              $('#id_figureindicator').prop('selectedIndex', 0);
+        var driver_checkbox = document.getElementById("id_check_driver");
+        var figureindicator_checkbox = document.getElementById("id_check_figureindicator");
+        var impact_checkbox = document.getElementById("id_check_impact");
+
+        $(':checkbox').not(this).prop('checked', false);
+        if (driver_checkbox.checked) {
+          $('[data-show-if=show-drivers]').removeClass('hide');
+          $('[data-show-if=show-impacts]').addClass('hide');
+          $('[data-show-if=show-figureindicators]').addClass('hide');
+          $('#id_figureindicator').prop('selectedIndex', 0);
+          $('#id_impact').prop('selectedIndex', 0);
           }
-          else if (figureindicator_checkbox.checked) {
-              $('[data-show-if=show-figureindicators]').removeClass('hide');
-              $('[data-show-if=show-drivers]').addClass('hide');
-              $('#id_driver').prop('selectedIndex', 0);
-          }
-          else {
+        else if (figureindicator_checkbox.checked) {
```

Comment:
I think this JS section is starting to be difficult to be managed :(. Can you please move the reused code to separate functions and call them for each check selector?